File: StuddyBuddy/src/main/java/com/licenta/StuddyBuddy/dto/ModuleRequest.java
package com.licenta.StuddyBuddy.dto;

import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ModuleRequest {
    private String courseId;
    private String moduleId;

    private String title;
    private String description;
    private String filePath;

}

File: angular-ui-studdy-buddy/src/app/interfaces/module-request.ts
import {Assignment} from "./course";

export interface ModuleRequest {
   courseId: string;
   moduleId: string;
   title: string;
   description: string;
   filePath: FilePath[];

  assignmentInstances: AssignmentInstance[];
}
export interface AssignmentInstance {
  assignmentInstanceId: string;
  assignment: Assignment;
  dueDate: Date;
}

export interface FilePath {
  fileDescriptionsId: string;
  filePath: string;
}
